generic/OpenMP_Support.F90

```diff
@@ -593,16 +593,17 @@ subroutine copy_callbacks(state, multi_states, rc)
        n_multi = size(multi_states)
        call get_callbacks(state, callbacks, _RC)
        _ASSERT(associated(callbacks), 'callbacks must be associated')
-       associate (e => callbacks%end())
-         iter = callbacks%begin()
-         do
-            wrapper => iter%second()
-            do i = 1, n_multi
-               call ESMF_MethodAdd(multi_states(i), label=iter%first(), userRoutine=wrapper%userRoutine, _RC)
-            end do
-            call iter%next()
-         end do
-       end associate
+       if (.not. callbacks%empty()) then
+          iter = callbacks%begin()
+          do
+             wrapper => iter%second()
+             do i = 1, n_multi
+                call ESMF_MethodAdd(multi_states(i), label=iter%first(), userRoutine=wrapper%userRoutine, _RC)
+             end do
+             call iter%next()
+             if (iter == callbacks%end()) exit
+          end do
+       end if
```

Comment:
The usual pattern is:
```suggestion
       associate( e => callbacks%end())
          iter = callbacks%begin()
          do while (iter /= e)
             wrapper => iter%second()
             do i = 1, n_multi
                call ESMF_MethodAdd(multi_states(i), label=iter%first(), userRoutine=wrapper%userRoutine, _RC)
             end do
             call iter%next()
          end do
       end associate
```
I.e., the containers are designed to do the "right" thing when empty. So no need to test for that.

Reply:
I thought `do while` is deprecated.